### ISSUE TITLE
False positive returned when multiple requestPermissions are called

### ISSUE THREAD

**Issue by zanemcca** [dpa99c/cordova-diagnostic-plugin#99]
I have two requests for permissions that overlap. The second request will return a `GRANTED` status without first opening the dialog. 

Here is my `adb logcat` output where you can see the second request returning immediately after it is called with a warning from `Activity` that is not being properly handled.

```
 14750 14830 V Diagnostic: Get authorisation status for android.permission.READ_EXTERNAL_STORAGE
07-19 12:36:39.807 14750 14830 V Diagnostic: Get authorisation status for android.permission.WRITE_EXTERNAL_STORAGE
07-19 12:36:39.811 14750 14830 V Diagnostic: Get authorisation status for android.permission.ACCESS_COARSE_LOCATION
07-19 12:36:39.811 14750 14830 V Diagnostic: Get authorisation status for android.permission.ACCESS_FINE_LOCATION
07-19 12:36:39.944 14750 14830 V Diagnostic: Get authorisation status for android.permission.READ_EXTERNAL_STORAGE
07-19 12:36:39.947 14750 14830 V Diagnostic: Get authorisation status for android.permission.WRITE_EXTERNAL_STORAGE
07-19 12:36:39.948 14750 14830 D Diagnostic: Requesting permission for android.permission.READ_EXTERNAL_STORAGE
07-19 12:36:39.948 14750 14830 D Diagnostic: Requesting permission for android.permission.WRITE_EXTERNAL_STORAGE
07-19 12:36:39.948 14750 14830 V Diagnostic: Requesting permissions
07-19 12:36:39.952   841  2107 I ActivityManager: START u0 {act=android.content.pm.action.REQUEST_PERMISSIONS pkg=com.google.android.packageinstaller cmp=com.google.android.packageinstaller/com.android.packageinstaller.permission.ui.GrantPermissionsActivity (has extras)} from uid 10864 on display 0
07-19 12:36:40.000   841  2107 I ActivityManager: Start proc 14948:com.google.android.packageinstaller/u0a19 for activity com.google.android.packageinstaller/com.android.packageinstaller.permission.ui.GrantPermissionsActivity
07-19 12:36:40.008 14750 14830 W PluginManager: THREAD WARNING: exec() call to Diagnostic.requestRuntimePermissions blocked the main thread for 64ms. Plugin should use CordovaInterface.getThreadPool().
07-19 12:36:40.014 14750 14830 V Diagnostic: Get authorisation status for android.permission.ACCESS_COARSE_LOCATION
07-19 12:36:40.014 14750 14830 V Diagnostic: Get authorisation status for android.permission.ACCESS_FINE_LOCATION
07-19 12:36:40.015 14750 14830 D Diagnostic: Requesting permission for android.permission.ACCESS_COARSE_LOCATION
07-19 12:36:40.015 14750 14830 D Diagnostic: Requesting permission for android.permission.ACCESS_FINE_LOCATION
07-19 12:36:40.015 14750 14830 V Diagnostic: Requesting permissions
07-19 12:36:40.015 14750 14830 W Activity: Can reqeust only one set of permissions at a time
07-19 12:36:40.016 14750 14830 V Diagnostic: Received result for permissions request id=191614
```

**Comment by dpa99c** [comment]
This is a result of the underlying native Android run-time permissions API being unable to handle multiple simultaneous requests. 

To resolve this, I would suggest either combining the multiple requests into a single request, or chaining the requests so the 2nd is not made until the 1st has completed.

I think what I'll do in the next patch release is add a semaphore flag to block multiple simultaneous requests being made, so attempting to make a request while another request has not resolved will result in the plugin error callback being invoked, rather than allowing the concurrent requests to be made. I will also make this clearer in the documentation.

**Comment by zanemcca** [comment]
I avoided calling them at the same time as a workaround. 

Instead of invoking an error could you queue the requests?

**Comment by dpa99c** [comment]
The Android runtime permissions API enables batch requesting of multiple permission groups. This allows a whole bunch of permissions to be requested with one flow of native dialogs. So this is the best way to go about it, as opposed to making multiple individual calls and having the plugin queue and chain them, because that results in more dialogs to hassle the user. For this reason, I will do as I first suggested and have the plugin prevent multiple concurrent permission requests by throwing an error if an unresolved one is in progress, but I'll expose an additional method that enables you to query if a request is already in progress.
